hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/codec/DecodingFacade.java

```diff
@@ -140,7 +141,13 @@ public static List<TokenID> decodeTokenIDsFromBytesArray(@NonNull final byte[][]
     public static AccountID convertLeftPaddedAddressToAccountId(
             final byte[] leftPaddedAddress, @NonNull final UnaryOperator<byte[]> aliasResolver) {
         final var addressOrAlias = Arrays.copyOfRange(leftPaddedAddress, ADDRESS_SKIP_BYTES_LENGTH, WORD_LENGTH);
-        return accountIdFromEvmAddress(aliasResolver.apply(addressOrAlias));
+        final var resolved = aliasResolver.apply(addressOrAlias);
+        if (!isMirror(resolved)) {
+            return AccountID.newBuilder()
+                    .setAlias(ByteStringUtils.wrapUnsafely(addressOrAlias))
+                    .build();
+        }
```

Comment:
I think it's actually impossible to get a non-mirror address here, since the [`AliasResolver` implementation](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java#L232) is [`StackedWorldUpdater.unaliased()`](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/HederaStackedWorldStateUpdater.java#L110)---and that always returns a mirror address (in particular, `0x00...00` if the given alias is invalid).

But if we _did_ somehow get a non-mirror address here, I think we should log at `WARN` and return `EntityNum.MISSING_NUM.toGrpcAccountId()`; there's a definite risk as written that we would create a record that mirror node couldn't handle. (They assume every `AccountID` in a record will be in `0.0.X` form.)

Reply:
I think that `aliasResolver` can return non-mirror address and that will be in the case where pass `addressOrAlias` which is a non-mirror address that does not exists (e.g. lazy create), then we will end up in `AliasManager.resolveForEvm` [here](https://github.com/hashgraph/hedera-services/blob/d1d2be6e67e219a746eb99e3a2ad4a0d1e2c8af9/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/AliasManager.java#L103) and we will return what was passed i.e. a non-mirror address.

The introduced changes are dealing with that case as far as I can remember, the initial commit was without this changes but there were [some integration tests failing](https://github.com/hashgraph/hedera-services/actions/runs/4282943227/jobs/7457856272) that led me down this road.

Reply:
Ah I get it now!

1. As you noted, the javadoc on `HederaStackedWorldStateUpdater.unaliased()` is a lie, it _can_ return non-mirror addresses. 🙈  
2. Before this PR, when a `TransferPrecompile` called the existence-aware form of `convertLeftPaddedAddressToAccountId()`, and it got back a non-mirror address, then the resulting cycle of transformations `0xab...ef1234 ➡️ ab.ef.1234 ➡️ 0.0.<alias=ab...ef1234>` preserved all the address information, and lazy-create "just worked."
3. But now, the cycle looks like `0xab...ef1234 ➡️ 0.0.1234 ➡️ 0.0.<alias=00...001234>`; and as a result `TransferPrecompile` throws an `INVALID_ALIAS_KEY` [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TransferPrecompile.java#L276), because it _thinks_ it has got a request to lazy-create a mirror address.

So you solve it and avoid the information loss by just going straight to `0.0.<alias=ab...ef1234>`. 😎 This is actually great, since on `develop` e.g. a call to the `associate()` precompile with a non-existent address will result in a synthetic `TransactionBody` like,
```
tokenAssociate {
  account {
    shardNum: 1234567
    realmNum: 1234567
    accountNum: 1234567
  }
  tokens {
    tokenNum: 1002
  }
}
  ➡️ receipt {
  status: INVALID_ACCOUNT_ID
}
```
...while on your branch it shows,
```
tokenAssociate {
  account {
    alias: "\000\022\326\207\000\000\000\000\000\022\326\207\000\000\000\000\000\022\326\207"
  }
  tokens {
    tokenNum: 1002
  }
}
  ➡️ receipt {
  status: INVALID_ACCOUNT_ID
}
```
...clearly much nicer!!

Would have been quite hard to notice this information loss if not for @dimitar-dinev's hard work on `LazyCreateThroughPrecompileSuite`. 🙏 

Ideally this would all be a _bit_ less confusing, of course...but that's for another day. 😅